### PR TITLE
Jp okta 437720

### PR DESCRIPTION
**Pull request by jaredperreault-okta**
## Description:
converts project structure to a `yarn workspace`

Docs: https://classic.yarnpkg.com/lang/en/docs/workspaces/

## PR Checklist

- [ ] Have you verified the basic functionality for this change?

- [ ] Added unit tests?

- [ ] Added e2e tests

- [ ] Did you follow our [Security Best Practices](https://oktawiki.atlassian.net/wiki/display/eng/Security+Best+practices)?

### Screenshot/Video:
N/A - purely project structure changes

### Issue:

- [OKTA-437720](https://oktainc.atlassian.net/browse/OKTA-437720)

### CODE REVIEW

File: test/e2e/package.json [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+  "name": "@okta/e2e",
+  "version": "1.0.0",
+  "main": "runner.js",
+  "private": true,
+  "scripts": {
+    "build": "yarn workspace @okta/okta-signin-widget build:release",
+    "test": "yarn build && node ./runner.js"
+  },
+  "dependencies": {
+    "@okta/okta-auth-js": "*",
```
Comment (shuowu) [inline test/e2e/package.json:11]: The widget is not always synced with the latest version of auth-js, let's bind auth-js to a specific major version with `^`.

File: package.json [hunk 2]
```diff
@@ -135,11 +125,13 @@
     "nodemon": "^2.0.7",
     "npm-run-all": "^4.1.5",
     "parse-ms": "^2.0.0",
+    "protractor": "^7.0.0",
```
Comment (shuowu) [inline package.json:128]: Do we still use protractor in anywhere?

File: test/app/package.json [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+  "name": "@okta/test.app",
+  "version": "1.0.0",
+  "main": "dist/main.bundle.js",
+  "private": true,
+  "scripts": {
+    "start": "webpack serve --config ./webpack.config.js",
+    "test": "echo \"Error: no test specified\" && exit 1"
+  },
+  "dependencies": {
+    "@okta/okta-auth-js": "*",
```
Comment (shuowu) [inline test/app/package.json:11]: bind to auth-js major version

File: package.json (hunk 2 above)
Comment (jaredperreault-okta) [reply package.json:128]: I think you're right, but I found it here https://github.com/okta/okta-signin-widget/blob/master/Gruntfile.js#L362.

I think this reference is dead code however

File: test/e2e/wdio.conf.js [hunk 4]
```diff
@@ -171,7 +175,7 @@ exports.config = {
     reporters: [
         'spec',
         ['junit', {
-            outputDir: 'build2/reports/junit',
+            outputDir: '../../build2/reports/junit',
```
Comment (aarongranick-okta) [inline test/e2e/wdio.conf.js:178]: i know somebody who would love to get rid of the `build2` directory. 😄

**Review by aarongranick-okta — APPROVED** [review]
Looks great!

**Review by shuowu-okta — APPROVED** [review]

**Review by shuowu-okta — APPROVED** [review]

**Pull request closed without merging**